lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java

```diff
@@ -237,4 +245,43 @@ void testWasb()
             assertThat(inputFile.exists()).isFalse();
         }
     }
+
+    @Test
+    @Override
+    public void testFileWithTrailingWhitespace()
+            throws IOException
+    {
+        if (useServerSideEncryptionWithCustomerKey()) {
+            abort("Azure requires decryption key to rename an encrypted file");
+            return;
+        }
+
+        super.testFileWithTrailingWhitespace();
+    }
+
+    @Test
+    @Override
+    public void testDirectoryExists()
+            throws IOException
+    {
+        if (useServerSideEncryptionWithCustomerKey()) {
+            abort("Azure requires decryption key to get properties of an encrypted file");
+            return;
+        }
+
+        super.testDirectoryExists();
+    }
+
+    @Test
+    @Override
+    protected void testRenameFile()
+            throws IOException
+    {
+        if (useServerSideEncryptionWithCustomerKey()) {
```

Comment:
It would be nicer if you assert it does not work in specific way so we know when/if behavior changes.